webpack/assets/javascripts/react_app/components/CPUCoresInput/CPUCoresInput.js

```diff
@@ -0,0 +1,81 @@
+import React, { useState, useCallback } from 'react';
+import PropTypes from 'prop-types';
+import { WarningTriangleIcon } from '@patternfly/react-icons';
+import { HelpBlock, FormGroup, Grid, Col, Row } from 'patternfly-react';
+import { translate as __ } from '../../common/I18n';
+import NumericInput from '../common/forms/NumericInput';
+import './cpuCoresInput.scss';
+import { noop } from '../../common/helpers';
+
+const CPUCoresInput = ({
+  label,
+  recommendedMaxValue,
+  maxValue,
+  defaultValue,
+  onChange,
+  minValue,
+}) => {
+  const [value, setValue] = useState(defaultValue);
+
+  const getValidationState = useCallback(() => {
+    if (value > recommendedMaxValue) return 'warning';
```

Comment:
It would be nice to be able to set the `recommendedMaxValue` or/and `maxValue` to `0` or `null` to disable their functionalists.